### DISCUSSION TITLE
Using RLE fonts - how to undo the run length encoding

### DISCUSSION THREAD

**Discussion by mcbpalisade** [Bodmer/TFT_eSPI#2505] [Q&A - Fonts]
From what I can tell these fonts are compressed in some fashion.  Is there a library call that undoes this?

If there isn't then I assume they aren't usable and I should continue my search for large numeric fonts.

Thank you.

**Comment by Bodmer** [comment]
The library decompresses each character as it is used. See this example for characters and sizes available in these builtin fonts.
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/320%20x%20240/RLE_Font_test

**Reply by mcbpalisade** [reply]
First, thanks for offering this information.

I examined that test code and didn't see anything I was really expecting. (I expect wrong a lot : - )  I don't see any "setFont" commands so I can't tell how different fonts are used.  I don't see anything tying these "drawString" commands to their source.  Can you explain this?

To be fair I didn't try your sketch because I use MVS and PlatformIO now and most *.ino files don't port without much difficulty.

Most cases that have bit maps in C arrays use the "drawBitmap" call.  I tried this but got screen gibberish.  No wonder as these are row compressed.  Given bit flash memory sizes these days I'd be happy with a 75 pixel high uncompressed C array of numerals only, if one of those exists or I can make one.

Thanks!  mcb
